fs.go

```diff
@@ -132,25 +132,24 @@ func checkDiskFree(diskPath string, minFreeDisk int64) (err error) {
 	return nil
 }
 
-// removeDuplicateVols - remove duplicate volumes.
-func removeDuplicateVols(vols []VolInfo) []VolInfo {
-	length := len(vols) - 1
-	for i := 0; i < length; i++ {
-		for j := i + 1; j <= length; j++ {
-			if vols[i].Name == vols[j].Name {
-				// Pick the latest volume, if there is a duplicate.
-				if vols[i].Created.Sub(vols[j].Created) > 0 {
-					vols[i] = vols[length]
-				} else {
-					vols[j] = vols[length]
-				}
-				vols = vols[0:length]
-				length--
-				j--
-			}
+func removeDuplicateVols(volsInfo []VolInfo) []VolInfo {
+	// Use map to record duplicates as we find them.
```

Comment:
Silently ignoring duplicate volumes is dangerous. It is better to write a function to detect and error out.